src/Database.cpp

```diff
@@ -127,11 +131,35 @@ void Database::checkMetaTable() {
     int dbVersion = sqlite3_column_int(stmt, 0);
     sqlite3_finalize(stmt);
 
-    if (dbVersion != DATABASE_VERSION) {
-        // we should be handling migrations here in the future
-        std::cout << "[FATAL] DB Version doesn't match Server Build" << std::endl;
+    if (dbVersion > DATABASE_VERSION) {
+        std::cout << "[FATAL] Server Build is incompatible with DB Version" << std::endl;
         exit(1);
     }
+
+    while (dbVersion != DATABASE_VERSION){
```

Comment:
Add a call to `std::filesystem::copy_file(settings::DBPATH, settings::DBPATH + ".old")` in a `try` before migrating?

https://en.cppreference.com/w/cpp/filesystem/copy_file